src/CoreWCF.BuildTools/src/OperationInvokerGenerator.Emitter.cs

```diff
@@ -0,0 +1,233 @@
+                    {
+                        _builder.AppendLine($"{indentor}return (null, outputs);");
+                    }
+                    else
+                    {
+                        _builder.AppendLine($"{indentor}return (result, outputs);");
+                    }
+                }
+                else
+                {
+                    if (operationContractSpec.Method.ReturnsVoid)
+                    {
+                        _builder.AppendLine($"{indentor}return new ValueTask<(object, object[])>((null, outputs));");
+                    }
+                    else
+                    {
+                        _builder.AppendLine($"{indentor}return new ValueTask<(object, object[])>((result, outputs));");
+                    }
+                }
+
+                indentor.Decrement();
+                _builder.AppendLine($"{indentor}}}");
+                _builder.AppendLine();
+                _builder.Append($"{indentor}public object[] AllocateInputs() => ");
+                if (inputParameterCount == 0)
+                {
+                    _builder.AppendLine("Array.Empty<object>();");
+                }
+                else
+                {
+                    _builder.AppendLine($"new object[{inputParameterCount}];");
+                }
+                _builder.AppendLine();
+                _builder.Append($"{indentor}private object[] AllocateOutputs() => ");
+                if (outputParameterCount == 0)
+                {
+                    _builder.AppendLine("Array.Empty<object>();");
+                }
+                else
+                {
+                    _builder.AppendLine($"new object[{outputParameterCount}];");
+                }
+                _builder.AppendLine();
+
+                _builder.AppendLine($"{indentor}[System.Runtime.CompilerServices.ModuleInitializer]");
+                _builder.Append($"{indentor}internal static void RegisterOperationInvoker() => ");
+                _builder.AppendLine($"CoreWCF.Dispatcher.DispatchOperationRuntimeHelpers.RegisterOperationInvoker(\"{operationContractSpec.Method.ToDisplayString()}\", new {operationInvokerTypeName}());");
+
+                indentor.Decrement();
+                _builder.AppendLine($"{indentor}}}");
+
+                indentor.Decrement();
+                _builder.AppendLine($"{indentor}}}");
+
+                string GetOperationInvokerTypeName()
+                {
+                    StringBuilder builder = new();
+                    builder.Append(operationContractSpec.Method!.ToDisplayString());
+                    builder.Replace(".", "_")
+                        .Replace("<", "_")
+                        .Replace(">", "_")
+                        .Replace("(", "_")
+                        .Replace(")", "_")
+                        .Replace("[", "_")
+                        .Replace("]", "_")
+                        .Replace(", ", "_")
+                        .Replace(" ", "_");
```

Comment:
I think it would be worth using something else to either replace comma or a space otherwise you'll get a collision between these two methods:
```
System.String Echo(System.String input, MyNameSpace.Ref refParam,Boolean b, out System.Int32 i);
System.String Echo(System.String input, ref System.Boolean b, out System.Int32 i);
```

Reply:
I found another collision. You will get a collision between these two I believe:
```c#
void Echo_bool(string value);
void Echo(bool isit, string value);
```

Reply:
Should we autogenerate the invoker class name to not correspond to the actual operation method to avoid this problem. Add a comment in the generated code to say which method it's for, although you can see that in the implementation method anyway, but it's easier if it's in a comment.

Reply:
I'm now generating class name like `OperationInvoker0`, `OperationInvoker2`, `OperationInvoker3`.. and a comment describes the method which is called.